app/Http/Controllers/DashboardController.php

```diff
@@ -67,185 +71,77 @@ public function dev_layout()
         
-        $data = $this->dashData($site, $location, $device);
+        //Store the active site, location, and device in a collection
+        $active_data = collect(['device' => $active_device, 'location' => $locations[0] ?? null, 'site' => $sites[0] ?? null]);
         
-        return response()->json([ 'active_device' => $data[0], 'devices' => $data[1], 'locations' => $data[2], 'sites' => $data[3] ]);
-    }
-    
-    /**
-     * Return database data about devices, sites, and locations
-     *
-     * @param int $location_id
-     * @return \Illuminate\Http\JsonResponse
-     */
-    public function locationChange($location_id)
-    {
-        //Get the given location
-        $location = Location::select('id', 'name', 'site_id')->findOrFail($location_id);
-        //Get the given site
-        $site = $location->site()->select('id', 'name')->firstOrFail();
-        //Get the given device if it is still at the same location
-        $device = Device::publicDashData()->where('location_id', '=', $location_id)->firstOrFail();
-        
-        $data = $this->dashData($site, $location, $device);
-        
-        return response()->json([ 'active_device' => $data[0], 'devices' => $data[1], 'locations' => $data[2], 'sites' => $data[3] ]);
-    }
-    
-    /**
-     * Return database data about devices, sites, and locations
-     *
-     * @param \Illuminate\Database\Eloquent\Model $site
-     * @param \Illuminate\Database\Eloquent\Model $location
-     * @param \Illuminate\Database\Eloquent\Model $device
-     * @return \Illuminate\Database\Eloquent\Collection
-     */
-    public function dashData($site, $location, $device)
-    {
-        //Get all the sites except for the current site ordered by name
-        $sites = Site::select('id', 'name')->orderBy('name', 'ASC')->get()->except($site->id);
-        
-        //Get all the locations for the given site except for the new current location ordered by name
-        $locations = $site->locations()->select('id', 'name', 'site_id')->orderBy('name', 'ASC')->get()->except($location->id);
-        
-        //Get all the devices that belong to the given location ordered by name
-        $devices = $location->devices()->publicDashData()->orderBy('name', 'ASC')->get();
-        
-        $active_device = collect([$device, $location, $site]);
-        
-        return collect([$active_device, $devices, $locations, $sites]);
-    }
-    
-    /**
-     * Open or close the given device
-     *
-     * @param  Request  $request
-     * @param Device $device
-     * @return \Illuminate\Database\Eloquent\Builder[]|\Illuminate\Database\Eloquent\Collection
-     */
-    public function updateCommand(Request $request, Device $device)
-    {
-        Validator::make($request->all(), [
-            'command' => 'required|string|max:1|in:1,2,3',
-        ])->validate();
+        //Get the total amount of pages for pagination
+        $page_count = ceil($deviceCount / $limit) - 1;
```

Comment:
Please us the builtins for pagination and check out the take() method.
https://laravel.com/docs/5.5/pagination